scripts/README.md

```diff
@@ -0,0 +1,175 @@
+
+### Invoke view function
+
+This is a fairly practical way to easily access on chain information from your shell. For example, from above, you might want to know how many seconds are remaining in the current batch befoer your withdraw request becomes available:
+
+```sh
+npx truffle exec build/common/scripts/invoke_view_function.js --network mainnet getSecondsRemainingInBatch
+```
+
+Some other potentially useful calls might be,
+
+```sh
+npx truffle exec build/common/scripts/invoke_view_function.js --network mainnet getCurrentBatchId
+npx truffle exec build/common/scripts/invoke_view_function.js --network mainnet tokenIdToAddressMap 1
+```
+
+Note that any additional arguments that should be passing into the view function should appear in order
+
+### Get Auction Elements
+
+This will return a human readable list of all orders meeting the simple search criteria provided via the following (optional) script arguments
+
+```sh
+npx truffle exec build/common/scripts/get_auction_elements.js --network rinkeby --expired true --covered false --tokens 0,1,2,3,4 --pageSize 50
+```
+
+Be warned that this may take a while!
+
+### Transitive Orderbook
+
+Computes a price at which the given amount of `sellToken` can be sold for `buyToken` given the orderbook of the batch that is currently collecting orders.
+
+```sh
+npx truffle exec build/common/scripts/transitive_orderbook.js --network rinkeby --sellToken 1 --buyToken 4 --sellAmount 1000
+```
+
+with additional optional agruments such as `hops` and `pageSize`
+
+## Involved Transactions
+
+### Add Token List
+
+This script is usually only needed upon initial deployment of a new BatchExchange contract to any network. It will then fetch a currated list of ERC20 tokens and register each of these for excahnge on the newly deployed contract.
+
+```sh
+npx truffle exec build/common/scripts/add_token_list.js
+```
+
+### Ensure OWL Liquidity
+
+This is meant to check and ensure that there are orders selling OWL at an appropriate price between all listed token on the exchange.
+
+```sh
+npx truffle exec build/common/scripts/ensure_owl_liquidity.js --network rinkeby
+```
+
+### Place Spread Orders
+
+This script is meant to be used with a token list of only stable tokens that are all valued essentially equally. This will place orders with configurable spread between each token pair listed. This script will inform the user of the orders that about to be placed and between which tokens and end with a user prompt confirming that the order placement transaction should be sent.
+
+Note that any undesired order placements can be cancelled using the `cancel_order` script along with the order Ids logged as a result of this script's execution. Furthermore, when default values for `validFrom` are used, the orders will not become valid for three batches (to allow ampel time for cancellation).
+
+```sh
+npx truffle exec build/common/scripts/place_spread_orders.js  --tokens 2,3,6 --network rinkeby
```

Comment:
I imagine these should be updated to use `yarn truffle exec`? I'd be fine if that were part of a separate PR.

Reply:
Will do this in separate PR.